util.go

```diff
@@ -0,0 +1,23 @@
+package codacytool
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+func parseJSONFile(fileLocation string, out interface{}) error {
```

Comment:
can we return:
```
error | Configuration
```
instead of filling up a pointer?

Reply:
We can't because we are parsing the JSON into an unknown interface{} to make the method generic.
By filling up a pointer, the `json.Unmarshal` will automatically parse into the struct type passed on the out param. This way, we can use this method to parse any json into any struct type.

Reply:
We are using this method in exactly two places, reading:
```
config := Configuration{}
err := parseJSONFile(fileLocation, &config)
```
and
```
tool := ToolDefinition{}
err := parseJSONFile(fileLocation, &tool)
```
and the only different part is the unmarshal, we can have the "read file to memory" functionality in a method and two specialized unmarshaller function to parse `Configuration` and `ToolDefinition`.

I wouldn't go this generic here